### PR TITLE
docs: clarify pull request review guidelines.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
- Move the guidelines from Homebrew/brew Maintainer Guide to Maintainer Guidelines as they apply to all repositories.
- Clarify the guidelines to make it clear that the default option should be to approve a PR, with or without comments.
- Note in the New Maintainer Checklist that the most important task for maintainers is reviewing pull requests.

### CODE REVIEW

File: docs/Maintainer-Guidelines.md
Comment (issyl0) [inline docs/Maintainer-Guidelines.md:35]:
```suggestion
It is (nearly) impossible to merge a PR without approval.
```

File: docs/Maintainer-Guidelines.md
Comment (samford) [inline docs/Maintainer-Guidelines.md:20]:
```suggestion
When reviewing a PR, use "approve", "approve with comments", "comment" or "request changes" when submitting based on the following guidelines:
```

File: docs/Maintainer-Guidelines.md
Comment (samford) [inline docs/Maintainer-Guidelines.md:28]:
```suggestion
  Save it for "if this PR is merged before I personally have had a chance to ✅ approve it: it seems very likely to cause user-visible problems".
```

File: docs/Maintainer-Guidelines.md
Comment (MikeMcQuaid) [reply docs/Maintainer-Guidelines.md:35]: iMessage leaking into PRs. This could have been way more embarrassing...

File: docs/New-Maintainer-Checklist.md
Comment (SMillerDev) [inline docs/New-Maintainer-Checklist.md:27]: Can we still do this? I don't think we can

Comment (MikeMcQuaid) [reply docs/New-Maintainer-Checklist.md:27]: updating to "opening PRs for"

File: docs/Maintainer-Guidelines.md
Comment (SMillerDev) [inline docs/Maintainer-Guidelines.md:27]: Does this also apply to core? Because while there are no user-visible problems from user binaries there I feel my usage of this change is appropriate for them.

File: docs/Maintainer-Guidelines.md
Comment (issyl0) [reply docs/Maintainer-Guidelines.md]: Ooof, yes.

File: docs/Maintainer-Guidelines.md
Comment (bevanjkay) [reply docs/Maintainer-Guidelines.md:27]: Also for the cask repos. I think that using "changes requests" indicates clearly to the PR that the PR cannot be merged until the changes are made. Other maintainers may dismiss the review if the changes have been made before an additional pass.

Comment (MikeMcQuaid) [reply docs/Maintainer-Guidelines.md:27]: @SMillerDev @bevanjkay Yes, this would apply everywhere (and we should have a consistent policy/treatment everywhere generally).

While it's possible to dismiss a "request changes": it feels culturally more friendly and saves everyone time and effort to save this for the case where "a specific person needs to rereview this PR before it should be merged".

Comment (SMillerDev) [reply docs/Maintainer-Guidelines.md:27]: Do we have another way to indicate that something is fundamentally wrong and the user should redo it?

Comment (MikeMcQuaid) [reply docs/Maintainer-Guidelines.md:27]: As stated here: "request changes" should be more "it's not worth making code changes until we've discussed if we even want this PR".

In general: people don't like "request changes". It is perceived as negative/aggressive and, in your case, it can be avoided by just not commenting and instead stating what you would like to be changed.

Comment (bevanjkay) [reply docs/Maintainer-Guidelines.md:27]: I certainly have never considered a request changes review as being negative, but a way of expressing that the changes are required before the PR can go ahead. 

I do find it very confusing that an approving review can have comments that require addressing before merging.

This conflicts somewhat with GitHub's own [documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#about-pull-request-reviews) about the feature, so it feels a bit confusing to me. 

> A review has three possible statuses:

> Comment: Submit general feedback without explicitly approving the changes or requesting additional changes.
> Approve: Submit feedback and approve merging the changes proposed in the pull request.
> Request changes: Submit feedback that must be addressed before the pull request can be merged.

File: docs/New-Maintainer-Checklist.md
Comment (EricFromCanada) [inline docs/New-Maintainer-Checklist.md:27]:
```suggestion

- reviewing pull requests (from users and other maintainers)
- triaging, debugging and fixing user-reported issues and applying
- opening PRs for widely used changes (e.g. version updates)

```
This is in a code block so `mdl` ignores it, but adding line breaks helps it stand out more.

**Review by EricFromCanada — APPROVED** [review]

File: docs/Maintainer-Guidelines.md
Comment (MikeMcQuaid) [reply docs/Maintainer-Guidelines.md:27]: > I certainly have never considered a request changes review as being negative, but a way of expressing that the changes are required before the PR can go ahead.

Sure, but I've had others reach out who do feel this way.

> I do find it very confusing that an approving review can have comments that require addressing before merging.

GitHub does not provide a "apply these suggestions and then you can merge".

The main issue here is that we regularly see cases where a PR is blocked for ~24 hours (longer over weekends) because someone has commented/requested changes but not ✅ so the PR _cannot be merged_ until after that's been done.

> This conflicts somewhat with GitHub's own [documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#about-pull-request-reviews) about the feature, so it feels a bit strange to me.

Their documentation is written for all of their customers. I can state from my 10 years experience of working there (on many timezone-distributed teams): my documentation above better matches how GitHub uses this feature than their documentation does 😅

Comment (bevanjkay) [reply docs/Maintainer-Guidelines.md:27]: I can understand where the delay can cause issues for sure.

In the past, at least in `homebrew-cask`, we have used a label to indicate that a response is required from the PR author or feedback required. It has been more recently that there has been a general movement toward "request changes". For non-trivial changes some PRs would bounce between being labelled "awaiting user reply" and "awaiting maintainer feedback", for visibility purposes.

**Review by apainintheneck — APPROVED** [review]
I think these are good general guidelines. We do something similar at work. Me and my boss are in the US and everyone else on the team is in Europe so the review cycle can be pretty long even with people shifting their hours a bit. I actually had a talk with a colleague maybe 6 months after joining where he gave me this advice.

There is always going to be a variance in when people think something is ready or good enough to merge. I think "approve with comments" is easier when you have some sort of rapport with the author of the PR since you are trusting them to address your feedback. Since only maintainers can merge in PRs, this seems reasonable.

File: docs/Maintainer-Guidelines.md
Comment (MikeMcQuaid) [reply docs/Maintainer-Guidelines.md:27]: @bevanjkay Ok, thanks for explaining.

Would you feel better about this if we stated to only avoid using "request changes" like this when reviewing work from other maintainers?

**Comment by MikeMcQuaid** [comment]
@apainintheneck Thanks ❤️

File: docs/Maintainer-Guidelines.md
Comment (bevanjkay) [reply docs/Maintainer-Guidelines.md:27]: I'm ok with the changes after hearing your reasoning, it creates less maintainer friction for merging the changes after the PR looks good. 

I would say in contrast to the Brew repo, that most code review suggestions in `homebrew-cask` are blockers. 

As an aside, we've shifted culturally to providing suggestions and coaching PRs to create better contributors, where in the past maintainers would just commit the changes. 

Thanks for discussing 👍🏻

**Review by issyl0 — APPROVED** [review]

**Review by bevanjkay — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]

**Review by timsutton — APPROVED** [review]
Thanks for this discussion, it's been very helpful context for me as a n00b here 😍

File: docs/Maintainer-Guidelines.md
Comment (krehel) [reply docs/Maintainer-Guidelines.md:27]: Throwing my bit in here, I echo @bevanjkay's comment.  I have used it in Cask for "this should be changed prior to approval" and typically are blockers, not suggestions.  I use comments for those.  And never considered it to be a requirement that the person requesting changes be the reviewing party.

I always took it at face value of "please make this change to meet [best practice | guideline]" so never considered to be bothered by it. 😃 

No argument - just noting where my thoughts are and have differed.  Happy to approve.

**Review by krehel — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Review by samford — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for review and comments, all!

**Pull request merged**
